File: suc/kotlin/014.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "012PodcastListen"
private const val UI_TYPE = "Media"
private const val STYLE_THEME = "Cold Gradient"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF38BDF8)
        val secondary = Color(0xFF0EA5E9)
        val tertiary = Color(0xFF0284C7)
        val background = Color(0xFFF0F9FF)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE0F2FE)
        val outline = Color(0xFFBFE0F4)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFDC2626)
        val onPrimary = Color(0xFF001F2F)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF082F49)
        val onSurface = Color(0xFF082F49)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.SemiBold)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(8.dp)
        val medium = RoundedCornerShape(12.dp)
        val large = RoundedCornerShape(20.dp)
    }
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
        val xxl = 32.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 4.dp, 2.dp, 0.12f)
        val level2 = ShadowSpec(3.dp, 8.dp, 4.dp, 0.14f)
        val level3 = ShadowSpec(6.dp, 12.dp, 6.dp, 0.16f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    var selectedTab by remember { mutableStateOf(0) }
    var showDialog by remember { mutableStateOf(false) }
    val tabs = listOf("Trending", "Following", "Recent")
    val podcasts = listOf("AI Insights", "Mindful Moments", "History Flash", "Tech Hour", "Design Flow")

    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Text(
                        "Podcasts",
                        style = MaterialTheme.typography.displayLarge,
                        color = MaterialTheme.colorScheme.onSurface
                    )
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = MaterialTheme.colorScheme.background)
            )
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .background(
                    Brush.verticalGradient(
                        listOf(
                            AppTokens.Colors.surfaceVariant,
                            AppTokens.Colors.background
                        )
                    )
                )
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            SingleChoiceSegmentedButtonRow(
                modifier = Modifier.fillMaxWidth()
            ) {
                tabs.forEachIndexed { index, label ->
                    SegmentedButton(
                        selected = selectedTab == index,
                        onClick = { selectedTab = index },
                        shape = AppTokens.Shapes.small,
                        colors = SegmentedButtonDefaults.colors(
                            activeContainerColor = AppTokens.Colors.primary,
                            inactiveContainerColor = AppTokens.Colors.surface,
                            activeContentColor = AppTokens.Colors.onPrimary,
                            inactiveContentColor = AppTokens.Colors.onSurface
                        )
                    ) {
                        Text(label)
                    }
                }
            }
            Column(
                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
            ) {
                podcasts.forEach {
                    Card(
                        shape = AppTokens.Shapes.medium,
                        colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Row(
                            modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column {
                                Text(it, style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface)
                                Text("30 min episode", style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.onSurface.copy(alpha = 0.7f))
                            }
                            Button(
                                onClick = { showDialog = true },
                                shape = AppTokens.Shapes.small,
                                colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.secondary, contentColor = AppTokens.Colors.onSecondary)
                            ) {
                                Text("Play")
                            }
                        }
                    }
                }
            }
        }
        if (showDialog) {
            AlertDialog(
                onDismissRequest = { showDialog = false },
                confirmButton = {
                    TextButton(onClick = { showDialog = false }) {
                        Text("OK", color = AppTokens.Colors.primary)
                    }
                },
                containerColor = AppTokens.Colors.surface,
                title = { Text("Playing Podcast", style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface) },
                text = { Text("Streaming audio...", style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.onSurface.copy(alpha = 0.8f)) }
            )
        }
    }


}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF0F9FF)
@Composable
fun PreviewScreen() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/014.swift
//
//  PodcastListenColdGradientApp.swift
//  iOS 16 compatible – full screen – Cold Gradient
//

import SwiftUI

@main
struct PodcastListenColdGradientApp: App {
    var body: some Scene {
        WindowGroup {
            PodcastColdRoot()
                .ignoresSafeArea()
                .statusBar(hidden: true)
        }
    }
}

// MARK: - Tokens
private enum PC {
    enum C {
        static let primary = Color(hex: 0x38BDF8)
        static let secondary = Color(hex: 0x0EA5E9)
        static let tertiary = Color(hex: 0x0284C7)
        static let background = Color(hex: 0xF0F9FF)
        static let surface = Color.white
        static let surfaceVariant = Color(hex: 0xE0F2FE)
        static let outline = Color(hex: 0xBFE0F4)
        static let success = Color(hex: 0x16A34A)
        static let warning = Color(hex: 0xF59E0B)
        static let error = Color(hex: 0xDC2626)
        static let onPrimary = Color(hex: 0x001F2F)
        static let onSecondary = Color.white
        static let onTertiary = Color.white
        static let onBackground = Color(hex: 0x082F49)
        static let onSurface = Color(hex: 0x082F49)
    }
}

struct PodcastColdRoot: View {
    @State private var selectedTab = 0
    @State private var showDialog = false
    @State private var selectedPodcast = ""
    
    private let tabs = ["Trending", "Following", "Recent"]
    private let podcasts = [
        "AI Insights",
        "Mindful Moments", 
        "History Flash",
        "Tech Hour",
        "Design Flow"
    ]

    var body: some View {
        GeometryReader { geo in
            let scale = min(geo.size.width / 360.0, geo.size.height / 640.0)
            
            VStack(spacing: 0) {
                // ===== Top Spacer =====
                Spacer()
                    .frame(height: 0 * scale) // 增加顶部空白
                    .padding(.bottom, 25)
                    
                // ===== Top App Bar =====
                ZStack {
                    Rectangle()
                        .fill(PC.C.background)
                        .frame(height: 790 * scale)
                    
                    Text("Podcasts")
                        .font(.system(size: 28 * scale, weight: .semibold))
                        .foregroundColor(PC.C.onSurface)
                }
                .frame(height: 60 * scale)


            
                // ===== Tab Selection =====
                HStack(spacing: 10 * scale) {
                    ForEach(0..<tabs.count, id: \.self) { index in
                        Button(action: {
                            selectedTab = index
                        }) {
                            HStack(spacing: 1 * scale) {
                                if selectedTab == index {
                                    Image(systemName: "checkmark")
                                        .font(.system(size: 13 * scale, weight: .bold))
                                }
                                Text(tabs[index])
                                    .font(.system(size: 14 * scale, weight: .medium))
                            }
                            .foregroundColor(selectedTab == index ? PC.C.onPrimary : PC.C.onSurface)
                            .frame(maxWidth: .infinity)
                            .frame(width:112)
                            .frame(height: 39 * scale)
                            .background(
                                RoundedRectangle(cornerRadius: 8 * scale)
                                    .fill(selectedTab == index ? PC.C.primary : PC.C.surface)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 8 * scale)
                                            .stroke(PC.C.outline, lineWidth: 1 * scale)
                                    )
                            )
                        }
                    }
                }
                .padding(.horizontal, 16 * scale)
                .padding(.vertical, 15 * scale)
                .background(
                    LinearGradient(
                        colors: [PC.C.surfaceVariant, PC.C.background],
                        startPoint: .top,
                        endPoint: .bottom
                    )
                )
                
                // ===== Podcast List =====
                ScrollView {
                    LazyVStack(spacing: 15 * scale) {
                        ForEach(podcasts, id: \.self) { podcast in
                            HStack {
                                VStack(alignment: .leading, spacing: 5 * scale) {
                                    Text(podcast)
                                        .font(.system(size: 20 * scale, weight: .medium))
                                        .foregroundColor(PC.C.onSurface)
                                    Text("30 min episode")
                                        .font(.system(size: 14 * scale))
                                        .foregroundColor(PC.C.onSurface.opacity(0.7))
                                }
                                
                                Spacer()
                                
                                Button(action: {
                                    selectedPodcast = podcast
                                    showDialog = true
                                }) {
                                    Text("Play")
                                        .font(.system(size: 14 * scale, weight: .medium))
                                        .foregroundColor(PC.C.onSecondary)
                                        .frame(width: 84 * scale, height: 39 * scale)
                                        .background(
                                            RoundedRectangle(cornerRadius: 8 * scale)
                                                .fill(PC.C.secondary)
                                        )
                                }
                            }
                            .padding(16 * scale)
                            .background(
                                RoundedRectangle(cornerRadius: 12 * scale)
                                    .fill(PC.C.surface)
                            )
                            .padding(.horizontal, 16 * scale)
                        }
                    }
                    .padding(.vertical, 12 * scale)
                }
                .background(PC.C.background)
                
                // ===== Bottom Spacer =====
                Spacer()
                    .frame(height: 20 * scale) // 增加底部空白
            }
            .background(
                LinearGradient(
                    colors: [PC.C.surfaceVariant, PC.C.background],
                    startPoint: .top,
                    endPoint: .bottom
                )
                .ignoresSafeArea()
            )
            .alert(isPresented: $showDialog) {
                Alert(
                    title: Text("Playing Podcast"),
                    message: Text("Streaming audio..."),
                    dismissButton: .default(Text("OK")) {
                        showDialog = false
                    }
                )
            }
        }
    }
}

// MARK: - Hex Color Extension
private extension Color {
    init(hex: UInt32) {
        self.init(
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0
        )
    }
}

// MARK: - Preview
struct PodcastColdRoot_Previews: PreviewProvider {
    static var previews: some View {
        PodcastColdRoot()
            .previewDevice("iPhone SE (3rd generation)")
    }
}
